### PR TITLE
replace sha256 with pedersen_hash in get_key

### PR DESCRIPTION
**Pull request by gballet**
Hashing has been entirely removed from the verkle tree EIP, in order to increase its friendliness to zks.arks. This PR implements the `perdersen_hash` function.

TODO: rebase on the main branch once #45 has been merged,

### CODE REVIEW

**Comment by gballet** [comment]
Currently, the polynomial evaluation function is so slow that some miner tests time out. I have to more than double the timeout for it to pass. I don't think that we can merge this before go-ipa uses multi-exponentiation.

File: trie/verkle.go [hunk 1]
```diff
@@ -118,6 +118,16 @@ func (trie *VerkleTrie) Commit(onleaf LeafCallback) (common.Hash, int, error) {
 	flush := make(chan verkle.VerkleNode)
 	go func() {
 		trie.root.(*verkle.InternalNode).Flush(func(n verkle.VerkleNode) {
+			if onleaf != nil {
+				if leaf, isLeaf := n.(*verkle.LeafNode); isLeaf {
+					for i := 0; i < verkle.NodeWidth; i++ {
+						if leaf.Value(i) != nil {
+							comm := n.ComputeCommitment().Bytes()
```
Comment (jwasinger) [inline trie/verkle.go:125]: Lots of redundant calls to `ComputeCommitment` here.  Maybe I am missing something?

Comment (gballet) [reply trie/verkle.go:125]: the commitment is cached, so it's not a problem. Unless I misunderstand your point?

Comment (jwasinger) [reply trie/verkle.go]: Oh whoops yeah you are right.

**Pull request merged**